src/app/shared/models/action-trigger.ts

```diff
@@ -1,5 +1,43 @@
+import { KeyAlias } from 'src/app/keyboard/value-objects/key-alias'
+
 export interface ActionTrigger<Data = unknown> {
   id: string
   actionId: string
   data: Data
 }
+
+export interface KeyboardTriggerData {
+  keys: string[]
+  actionArguments?: unknown
+}
+
+export interface KeyboardAndSelectionTriggerData extends KeyboardTriggerData {
+  selected?: boolean
+}
+
+export interface CreateActionTrigger<Data = unknown> {
+  actionId: string
+  data: Data
+}
+
+export enum ActionTriggerSortKeys {
+  ACTION_ID_A_Z = 'ACTION_ID_A_Z',
+  ACTION_ID_Z_A = 'ACTION_ID_Z_A',
+  SHORTCUT_A_Z = 'SHORTCUT_A_Z',
+  SHORTCUT_Z_A = 'SHORTCUT_Z_A',
+}
```

Comment:
The purpose of this enum is to determine sort order. That doesn't seem like the enum needs to be tied to a specific attribute on specific interfaces.
Something like
```
export enum SortOrder {
  ALPHABETICAL = 'ALPHABETICAL',
  REVERSE_ALPHABETICAL = 'REVERSE_ALPHABETICAL'
}
```
should be useable for most arrays that should be sorted after a string value.

Reply:
I split the logic into 2 Enums and load them separately in the component. I don't know if this is meant in the comment or if we want to change the whole UI. I look forward to the next guidelines if we want a comprehensive change.

```
public readonly sortActionsTriggers: SofieDroppdownOptions[] = [
    { key: `${ActionTriggerSortKeys.ACTION}_${SortOrder.ALPHABETICAL}`, label: $localize`action-triggers-sort.action-label-a-z.label` },
    { key: `${ActionTriggerSortKeys.ACTION}_${SortOrder.REVERSE_ALPHABETICAL}`, label: $localize`action-triggers-sort.action-label-z-a.label` },
    { key: `${ActionTriggerSortKeys.SHORTCUT}_${SortOrder.ALPHABETICAL}`, label: $localize`action-triggers-sort.shortcut-a-z.label` },
    { key: `${ActionTriggerSortKeys.SHORTCUT}_${SortOrder.REVERSE_ALPHABETICAL}`, label: $localize`action-triggers-sort.shortcut-z-a.label` },
  ]
```